plc4go/internal/opcua/SecureChannel.go

```diff
@@ -1162,13 +1175,13 @@ func (s *SecureChannel) onDiscoverGetEndpointsRequest(ctx context.Context, codec
 					s.log.Debug().Type("type", messagePDU).Msg("Not relevant")
 					return false
 				}
-				return messageResponse.GetRequestId() == transactionId
+				return messageResponse.GetMessage().GetSequenceHeader().GetRequestId() == transactionId
 			},
 			func(message spi.Message) error {
 				opcuaAPU := message.(readWriteModel.OpcuaAPU)
 				messagePDU := opcuaAPU.GetMessage()
 				messageResponse := messagePDU.(readWriteModel.OpcuaMessageResponse)
-				readBuffer := utils.NewReadBufferByteBased(messageResponse.GetMessage(), utils.WithByteOrderForReadBufferByteBased(binary.LittleEndian))
+				readBuffer := utils.NewReadBufferByteBased(messageResponse.(readWriteModel.BinaryPayloadExactly).GetPayload(), utils.WithByteOrderForReadBufferByteBased(binary.LittleEndian))
```

Comment:
is it ensured that this is always `readWriteModel.BinaryPayloadExactly`?. Otherwise this part will panic